<launch>
    <!-- test environment -->
    <arg name="environment" default="$(optenv ENVIRONMENT AWS)"/>
    <!-- Test timeout -->
    <arg name="sim_timeout_seconds" default="$(optenv ROBOT_LOCALIZATION_WORLD_CHANGES_TEST_SIM_TIMEOUT_SECONDS 300)"/>
    <!-- waiting until the robot is spawned and stabilized-->
    <arg name="start_time_seconds" default="$(optenv ROBOT_LOCALIZATION_WORLD_CHANGES_TEST_START_TIME_SECONDS 10)" />
    <!-- model name -->
    <arg name="model_name" default="$(optenv MODEL_NAME robot)"/>
    <!-- goal position tolerance -->
    <arg name="localization_position_tolerance" default="$(optenv ROBOT_LOCALIZATION_WORLD_CHANGES_TEST_POSITION_TOLERANCE 0.2)"/>
    <!-- goal orientation tolerance -->
    <arg name="localization_orientation_tolerance" default="$(optenv ROBOT_LOCALIZATION_WORLD_CHANGES_TEST_ORIENTATION_TOLERANCE 0.2)"/>
    <!-- service name used in navigation stack adapter -->
    <arg name="goal_service" default="goal_service"/>
    <!-- mode for route manager(dynamic, inorder or random) -->
    <arg name="mode" default="$(optenv ROBOT_LOCALIZATION_WORLD_CHANGES_TEST_MODE dynamic)"/>
    <!-- mode for objects move manager(move or delete) -->
    <arg name="objects_manager_mode" default="$(optenv ROBOT_LOCALIZATION_WORLD_CHANGES_TEST_OBJECTS_MANAGER_MODE move)"/>
    <!-- map frame id -->
    <arg name="map_frame_id" default="map"/>
    <!-- topic, where goal result is published -->
    <arg name="goal_reached_topic_name" default="goal_reached"/>
    <!-- topic, where the pose estimated by a localization tool is published -->
    <arg name="localization_topic_name" default="/amcl_pose"/>
    <!-- amount of successfully navigated points -->
    <arg name="navigation_success_count" default="$(optenv ROBOT_LOCALIZATION_WORLD_CHANGES_TEST_NAVIGATION_SUCCESS_COUNT 5)"/>
    <!-- amount of objects that should be moved in the world -->
    <arg name="objects_count" default="$(optenv ROBOT_LOCALIZATION_WORLD_CHANGES_TEST_OBJECTS_COUNT 5)"/>

    <node pkg="mp_localization_tests" type="localization_world_changes_test_node.py" name="localization_world_changes_test_node" output="screen" required="true">
        <param name="sim_timeout_seconds" type="int" value="$(arg sim_timeout_seconds)" />
        <param name="start_time_seconds" type="int" value="$(arg start_time_seconds)" />
        <param name="model_name" type="string" value="$(arg model_name)" />
        <param name="environment" type="string" value ="$(arg environment)"/>
        <param name="goal_service" type="string" value="$(arg goal_service)"/>
        <param name="mode" type="string" value="$(arg mode)"/>
        <param name="objects_manager_mode" type="string" value="$(arg objects_manager_mode)"/>
        <param name="map_frame_id" type="string" value="$(arg map_frame_id)"/>
        <param name="localization_position_tolerance" type="double" value="$(arg localization_position_tolerance)"/>
        <param name="localization_orientation_tolerance" type="double" value="$(arg localization_orientation_tolerance)"/>
        <param name="goal_reached_topic_name" type="string" value="$(arg goal_reached_topic_name)"/>
        <param name="localization_topic_name" type="string" value="$(arg localization_topic_name)"/>
        <param name="navigation_success_count" type="int" value="$(arg navigation_success_count)"/>
        <param name="objects_count" type="int" value="$(arg objects_count)"/>
    </node>
</launch>